src/js/main.js

```diff
@@ -141,27 +155,61 @@ Site.Player = {
     $(window).on('updatedyoutubelist', function(event, data) {
 
       // Update list cache
-      _this.list = data.youtubeIds;
+      _this.setVideosList(data.youtubeIds);
     });
+
   },
 
   initYoutube: function() {
     var _this = this;
 
     // Init youtube player inside #player-container
-    _this.$player = new YT.Player('player-iframe', {playerVars: _this.playerOptions});
+    _this.player = new YT.Player('player-iframe', {
+      playerVars: _this.playerOptions,
+    });
+
+    _this.player.addEventListener('onStateChange', _this.handleVideoStateChange.bind(this));
 
   },
 
-  playVideo: function(videoId, list) {
+  handleVideoStateChange: function(event) {
+    var _this = this;
+
+    // 0 = Video ended
+    if(event.data === 0) {
+      // Check if there's list
+      if(_this.list) {
+        _this.nextVideo();
+      }
+    }
+  },
+
+  playVideo: function(videoId) {
     var _this = this;
 
     // Play video
-    _this.$player.loadVideoById(videoId);
+    _this.player.loadVideoById(videoId);
 
-    // Update list cache
-    if(list) {
-      _this.list = list;
+  },
+
+  nextVideo: function() {
+    var _this = this;
+
+    // Get current video ID
+    var currentVideo =  _this.player.getVideoData();
+    currentVideo = _this.list.indexOf(currentVideo.video_id);
+
+    // Check if theres more videos to play
+    if (_this.list.length > currentVideo + 1) {
+
+      // Get next video ID
+      var nextVideo = _this.list[currentVideo + 1];
+
+      // Play next video
+      _this.playVideo(nextVideo);
+    } else {
+      // TODO Close video
+      console.log('No more videos');
```

Comment:
I don't think so. This behavior is something we should talk about and define.
Should it loop?
Should it close the video?
Should it keep playing thru the excluded videos and reset the filters?
What do?

Made a card in trello :)